### PR TITLE
Finished load.py script

### PR DESCRIPTION
**Pull request by OscarB28**
Finished load.py

### CODE REVIEW

File: web_scraper/extract.py [hunk 1]
```diff
@@ -38,11 +38,11 @@ def load_all_data(conn: extensions.connection) -> set:
...
         url_list = [url[0] for url in urls]
         return convert_to_set(url_list)
-    
+
 
 def convert_to_set(urls: list[str]) -> set:
     """Turns the list into a set to remove duplicate entries."""
-    
+
     return set(urls)
```
Comment (BenBeard115) [inline web_scraper/extract.py:46]: Does this need to be it's own function?

File: web_scraper/extract.py [hunk 2]
```diff
@@ -84,15 +84,15 @@ def scrape_all_urls(urls: set) -> None:
 
 
 if __name__ == "__main__":
-    
+
     load_dotenv()
 
     startup = perf_counter()
     print("Loading data...")
```
Comment (BenBeard115) [inline web_scraper/extract.py:91]: Would these print lines be more useful as logs?

File: web_scraper/load.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from bs4 import BeautifulSoup
+from dotenv import load_dotenv
+from urllib.request import urlopen
+import requests
+
+
+def sanitise_filename(filename: str) -> str:
+    """Remove special characters from filename."""
+
+    return re.sub(r'[^\w\s.-]', ' ', filename)
```
Comment (BenBeard115) [inline web_scraper/load.py:17]: Nice

File: web_scraper/extract.py (hunk 1 above)
Comment (shivani-63) [reply web_scraper/extract.py:46]: We added this as a separate function for testing

File: web_scraper/load.py [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    """Uploads a file to the specified S3 bucket."""
+
+    try:
+        s3_client.upload_file(filename, bucket, key)
+
+    except Exception as e:
+        print('Unable to upload file. Please check details!')
+
+
+def upload_to_s3(s3_client: client, url: str, html_filename_temp: str, css_filename_temp: str):
```
Comment (BenBeard115) [inline web_scraper/load.py:48]: Very well structured.

File: web_scraper/load.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+    response = requests.get(url)
+    response.raise_for_status()
+
+    soup = BeautifulSoup(response.content, 'html.parser')
+    title = sanitise_filename(soup.title.text.strip())
+    html_content = str(soup)
+    css_content = '\n'.join([style.text for style in soup.find_all('style')])
+    html_filename_temp = f"{title}_html.html"
+    css_filename_temp = f"{title}_css.css"
```
Comment (BenBeard115) [inline web_scraper/load.py:82]: Could benefit from a little formatting for readability.

**Review by BenBeard115 — APPROVED** [review]
Looks good other than a few changes that you might consider beneficial.

**Pull request merged**